Lib/imaplib.py

```diff
@@ -1279,6 +1389,251 @@ def print_log(self):
                 n -= 1
 
 
+class Idler:
+    """Iterable context manager: start IDLE & produce untagged responses
+
+    An object of this type is returned by the IMAP4.idle() method.
+    It sends the IDLE command when activated by the 'with' statement, produces
+    IMAP untagged responses via the iterator protocol, and sends DONE upon
+    context exit.
+
+    Iteration produces (type, datum) tuples.  They slightly differ
+    from the tuples returned by IMAP4.response():  The second item in the
+    tuple is a single datum, rather than a list of them, because only one
+    untagged response is produced at a time.
+
+    Note: The name and structure of this class are internal interfaces,
+    subject to change.  Calling code can rely on its context management,
+    iteration, and public method to remain stable, but should not
+    subclass, instantiate, or otherwise directly reference the class.
+    """
+
+    def __init__(self, imap, dur=None):
+        if 'IDLE' not in imap.capabilities:
+            raise imap.error("Server does not support IDLE")
```

Comment:
Per Guido's suggestions:
```suggestion
            raise imap.error("Server does not support IMAP4 IDLE")
```

Reply:
Are you sure about this one? This would make it inconsistent with imaplib's existing "Server does not support..." exception message, and given the context, I think it's safe to assume the programmer who reads it knows it's not talking about the IDLE editor.

I'll make the change if you're sure you want it, though.

Reply:
My main worry is if this error were to come up in an application where imaplib isn't in the context. For example, if you're using some library that uses imaplib under the hood, and you see `Server does not support IDLE` in logs, then someone could possibly interpret it as the editor. I'll let you make the judgement call though.

Reply:
Fair enough. I wouldn't think confusion in this scenario likely, but in any case, the change does no harm. :)